lib/tasks/check-modulePrefix-mismatch.js

```diff
@@ -0,0 +1,13 @@
+import { getAppName } from '../utils/get-app-name.js';
+import { readFile } from 'node:fs/promises';
+
+export async function checkModulePrefixMisMatch() {
+  const packageJSON = JSON.parse(await readFile('package.json', 'utf-8'));
+  const modulePrefix = await getAppName();
+
+  if (packageJSON.name === modulePrefix) return;
+
+  throw new Error(
+    `Unexpected modulePrefix mismatch! package.json#name is ${packageJSON.name}, but modulePrefix is ${modulePrefix}. These two values should match`,
+  );
```

Comment:
The task looks clear to me; I have only one question whose answer could turn into a PR description. What kind of side effects do we expect if we meet the error case, why is it the codemod's problem?

Reply:
It's the codemod's problem because in vite, _nothing works at all_ ( 🙈! ) when these fields don't match.

It's a weird case to support, and I don't really think we need to support it.
(I say this as someone who has ~ 10 projects that this needs fixing in)

Reply:
Oww, ok, I can reproduce the problem, but was it really working before the upgrade to Vite? 🤔 I run into serious issues when I try a different `packageJSON.name` in a classic app. But anyway, since it's a different error, and since you ran into the problem, I would say it's fine to support this here 👍

Reply:
>  but was it really working before the upgrade to Vite? 

it did/does!

> in a classic app

you have to also set name in ember-cli-build.js, too